R/helper_functions.R

```diff
@@ -170,41 +227,83 @@ check_bestpreds_obs <- function(modelLibrary){
 #' @description Check that a list of predictions from a caretList are valid
 #'
 #' @param modelLibrary a list of predictins from caret models
-check_bestpreds_preds <- function(modelLibrary){
-  #TODO: ID which model(s) have bad preds
-  #TODO: Regression models should be numeric, classification models should have numeric class probs
+check_bestpreds_preds <- function(modelLibrary) {
+  # TODO: ID which model(s) have bad preds
+  # TODO: Regression models should be numeric, classification models should have numeric class probs
   pred <- lapply(modelLibrary, function(x) x[["pred"]])
   names(pred) <- names(modelLibrary)
   classes <- unique(sapply(pred, class))
   check <- length(classes)
-  if(check != 1){
+  if (check != 1) {
     stop(
       paste0(
         "Component models do not all have the same type of predicitons.  Predictions are a mix of ",
-        paste(classes, collapse=", "),
-        ".")
+        paste(classes, collapse = ", "),
+        "."
+      )
     )
   }
   return(invisible(NULL))
 }
 
+#' @title Check multiclass excluded level
+#' @description Verifies that the multiclass excluded level is
+#' within the range of the number of classes.
+#'
+#' @param excluded_level the level to exclude
+#' @param num_classes the number of classes
+check_multiclass_excluded_level <- function(excluded_level, num_classes) {
+  if (excluded_level < 1 || excluded_level > num_classes) {
+    warning(paste0(
+      "The excluded level must be between 1 and the number of classes (",
+      num_classes,
+      "). ",
+      "Provided value was ",
+      excluded_level,
+      ". ",
+      "\nThis value can be changed using setMulticlassExcludedLevel(). ",
+      "\nAttempting to train a model with all classes included."
+    ))
+  }
+}
+
+#####################################################
+# caretEnsemble check functions
+#####################################################
+#' @title Check binary classification
+#' @description Check that the problem is a binary classification problem
+#'
+#' @param list_of_models a list of caret models to check
+check_binary_classification <- function(list_of_models) {
+  if (is.list(list_of_models) && length(list_of_models) > 1 &&
+    is(list_of_models[[1]], "train") &&
```

Comment:
I think you should loop over `list_of_models` and check each model, rather than only checking the first model.

Reply:
OK, I will do it